kolibri/core/errorreports/models.py

```diff
@@ -47,25 +55,58 @@ def allow_migrate(self, db, app_label, model_name=None, **hints):
 
 
 class ErrorReports(models.Model):
-    error_from = models.CharField(max_length=10, choices=POSSIBLE_ERRORS)
+    category = models.CharField(max_length=10, choices=POSSIBLE_ERRORS)
     error_message = models.CharField(max_length=255)
     traceback = models.TextField()
     first_occurred = models.DateTimeField(default=timezone.now)
     last_occurred = models.DateTimeField(default=timezone.now)
-    sent = models.BooleanField(default=False)
-    no_of_errors = models.IntegerField(default=1)
+    reported = models.BooleanField(default=False)
+    events = models.IntegerField(default=1)
+    release_version = models.CharField(
+        max_length=64, default=".".join(map(str, VERSION[:2]))
+    )
+    installation_type = models.CharField(max_length=64, blank=True)
+    context_frontend = JSONField(
```

Comment:
I am not convinced about separate fields for frontend and backend context - it suggests that when we add reporting for the tasks we will have to have a third field for context for those and so on.

I think it would be better to have a single context field, together provide the specific validation we can either make a more intelligent validor that chooses the correct JSONSchema to apply, or we can do the validation outside of model validation. I prefer the former as it gives us a better guarantee.

However, to conditionalise the JSONSchema validation on the category (which I assume is backend, frontend etc) this would need to be done in a model method rather than in a field validator.

Can do it inside the clean method and then call that inside a save method before calling the save super.